### ISSUE TITLE
g:minisnip_finalstartdelim conflicts with Go templates

### ISSUE THREAD

**Issue by arp242** [joereynolds/vim-minisnip#45]
The syntax for [Go templates](https://golang.org/pkg/text/template/) is:

    {{.Variable}}

You can control leading and trailing whitespace by adding `-`:

    {{- .Variable}}
    {{.Variable -}}
    {{- .Variable -}}

That last variant conflicts with minisnip's `g:minisnip_finalstartdelim`, which is rather annoying.

I know this variable can be changed, but it seems to me that using a better default would be, well, better. Something like `{{|-` should be much more unique.

On the other hand, changing this would break compatibility for people with existing templates :-/

Not sure what the best course of action it. Either way, I thought I'd report it here.

**Comment by sigzero** [comment]
I would not be in favor of this change. The variable can be changed if you don't like it and a "better default" is highly subjective.

**Comment by arp242** [comment]
It's not about "liking", it's about conflicting with a reasonably common language. That's not a "highly subjective" matter, and I would argue that anything that *doesn't* conflict with anything common is objectively better as a simple pragmatical matter.

**Comment by sigzero** [comment]
You can change them to not conflict so no change to minisnip is needed. It is about liking.

**Comment by arp242** [comment]
Yes, I am aware. But then everyone writing Go templates will run in to this problem (which is rather confusing) and then have to change it. A default that no one has to change is better. The biggest issue is one of backwards compatibility, and whether it's worth the change.

**Comment by sigzero** [comment]
And when another language comes out with a template that uses whatever minisnip changes its default to then the discussion happens all over again. If I could vote, it would be no. It works for everyone now and allows the default to change when it needs to do so.

**Comment by sigzero** [comment]
Just as a note, not sure Joe is updating minisnip these days but I could be wrong.

**Comment by arp242** [comment]
There are choices that are much less likely to conflict with anything. `{{|-`, `{{||--`, or something else.

> Just as a note, not sure Joe is updating minisnip these days but I could be wrong.

Yeah, I don't know either. I wanted to document the issue in either case, even if no action is taken.

**Comment by joereynolds** [comment]
> Just as a note, not sure Joe is updating minisnip these days but I could be wrong.

Not recently, I'm busy with other projects at the moment and it conflicted in some ways with coc.nvim so I stopped using it. I'm not sure if that's still the case though.

I do intend to get back on it one day, but I have other things going on at the moment, I appreciate the enthusiasm though! :D
